### PR TITLE
Add nanosecond timestamp + workflow info to environment events

### PR DESCRIPTION
**Pull request by teo**
This PR does 2 things:

* [OCTRL-917](https://its.cern.ch/jira/browse/OCTRL-917) - adds a new `WorkflowTemplateInfo` field to all `EnvironmentEvent`s pushed to Kafka, that includes the public trait of the workflow template as requested, plus the name, description and workflow path.
* [OCTRL-918](https://its.cern.ch/jira/browse/OCTRL-918) - in addition to the millisecond timestamp, adds a nanosecond timestamp field `timestampNano` to all events pushed to Kafka. The timestamp itself is identical to the existing ms timestamp, it is just outputted with higher precision.

@martinboulais @graduta

### CODE REVIEW

File: core/environment/utils.go
Comment (knopers8) [inline core/environment/utils.go:38]:
```suggestion
type WorkflowInfo struct {
```
the name could be updated.

**Review by knopers8 — APPROVED** [review]
good for me, but I will wait for Martin or George's confirmation before merging.

**Comment by martinboulais** [comment]
@knopers8 So if I understood correctly, if `event.environmentEvent.workflowTemplateInfo.public` is false, it should not be stored in Bookeeping?

**Comment by knopers8** [comment]
@martinboulais I would even go further, I would require `event.environmentEvent.workflowTemplateInfo.name == "readout-dataflow"`. In the ECS GUI you can notice also the template `qc-server-workflow` and as far as I know, it also should not be recognized by BKP (and historically was not).

**Comment by martinboulais** [comment]
This seems a bit too hacky to me, no? I don't really like the hardcoded values like this, especially when it's from another service.
Why not simply marking the `public` to false in the workflow template?

**Comment by knopers8** [comment]
What I think would be the best is to add another field/variable in the templates such as `bookkeeped` and  `WorkflowInfo` would include it.

**Comment by vascobarroso** [comment]
@knopers8 @martinboulais I think the suggestion to add a new field makes a lot of sense. The current `WorkflowPublicInfo` value is to let the AliECS GUI know if it should be shown or not in the env list. Storing in bookkeeping is similar but not fully equivalent, so a new `WorkflowStoreInBookkeeping` or something like that makes sense.

**Comment by martinboulais** [comment]
@vascobarroso @knopers8 I agree that adding a new field makes quite some sense to me.

File: core/environment/utils.go
Comment (knopers8) [reply core/environment/utils.go:38]: ah ok, i think i understand what "public" refers to now.

**Comment by teo** [comment]
@knopers8 `bookkept` is now a variable as discussed with @martinboulais, but the rest of this PR still applies.

**Review by knopers8 — APPROVED** [review]
thank you

**Pull request merged**
